### PR TITLE
Group ball points by graph connectivity to improve ransac performance

### PR DESCRIPTION
**Pull request by TrentHouliston**

### CODE REVIEW

File: module/vision/BallDetector/data/config/BallDetector.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ green_ratio_threshold: 0.1
 green_radial_samples: 12
```
Comment (JakeFountain) [inline module/vision/BallDetector/data/config/BallDetector.yaml]: probably want to change the config paramters to be lower: 1 model, 30 iterations, etc.

File: module/vision/BallDetector/src/BallDetector.cpp [hunk 2]
```diff
@@ -203,207 +205,265 @@ namespace vision {
...
+                                      it2 = imagePoints.erase(it2);
+                                  }
+                                  else {
+                                      ++it2;
+                                  }
+                              }
+                          }
+                      }
+
+                      log("Groups: ", screenPointGroups.size());
```
Comment (JakeFountain) [inline module/vision/BallDetector/src/BallDetector.cpp]: make logs optional with debug ball throwouts or something

File: module/vision/BallDetector/src/BallDetector.cpp [hunk 3]
```diff
@@ -203,207 +205,265 @@ namespace vision {
...
+                      int i      = 1;
+                      // Loop through the groups and perform ransac on each
+                      for (const auto& group : screenPointGroups) {
+                          if (group.size() < MINIMUM_POINTS_FOR_CONSENSUS) {
+                              continue;
+                          }
+                          // This holds our points that may be a part of the ball
+                          std::vector<arma::vec3> ballPoints;
+
+                          log("group: ", i++, " size = ", group.size());
```
Comment (JakeFountain) [inline module/vision/BallDetector/src/BallDetector.cpp]: logs again

File: module/vision/BallDetector/src/BallDetector.cpp [hunk 4]
```diff
@@ -322,89 +309,165 @@ namespace vision {
...
+                               *                 MEASUREMENTS                 *
+                               ************************************************/
+
+                              Ball b;
+                              b.visObject.sensors = image.sensors;
+
+                              // Work out how far away the ball must be to be at the distance it is from the camera
+                              arma::vec3 width_rBCc = ballCentreRay * widthDistance;
+
+                              arma::vec3 rBCc = (width_rBCc);
```
Comment (Bidski) [inline module/vision/BallDetector/src/BallDetector.cpp]: Why is this here? Why not just `arma::vec3 rBCc = ballCentreRay * widthDistance'`?

Comment (TrentHouliston) [reply module/vision/BallDetector/src/BallDetector.cpp]: I believe previously it was the average of the two measurements, now it seems it's just one

**Comment by TrentHouliston** [comment]
This is based on the old ball detector
